tests/statespace/models/structural/components/test_cycle.py

```diff
@@ -0,0 +1,288 @@
+    """Test multivariate cycle component with innovations and estimated cycle length."""
+    cycle = st.CycleComponent(
+        name="cycle",
+        estimate_cycle_length=True,
+        innovations=True,
+        dampen=True,
+        observed_state_names=["data_1", "data_2", "data_3"],
+    )
+    params = {
+        "cycle": np.array([[1.0, 1.0], [2.0, 2.0], [3.0, 3.0]], dtype=config.floatX),
+        "cycle_length": 12.0,
+        "cycle_dampening_factor": 0.95,
+        "sigma_cycle": np.array([0.5, 1.0, 1.5]),  # different innov variances per var
+    }
+    x, y = simulate_from_numpy_model(cycle, rng, params)
+
+    cycle.build(verbose=False)
+    _assert_basic_coords_correct(cycle)
+
+    assert cycle.coords["cycle_state"] == ["cycle_Cos", "cycle_Sin"]
+    assert cycle.coords["cycle_endog"] == ["data_1", "data_2", "data_3"]
+
+    assert cycle.k_endog == 3
+    assert cycle.k_states == 6  # 2 states per variable
+    assert cycle.k_posdef == 6  # 2 innovations per variable
+
+    # Check that the data has the expected shape
+    assert y.shape[1] == 3  # 3 variables
+
+    # Check that each variable shows some variation (due to innovations)
+    for i in range(3):
+        assert np.std(y[:, i]) > 0
+
+    # check design, transition, selection & state_cov matrices
+    Z, T, R, Q = pytensor.function(
+        [
+            cycle._name_to_variable["cycle_length"],
+            cycle._name_to_variable["cycle_dampening_factor"],
+            cycle._name_to_variable["sigma_cycle"],
+        ],
+        [
+            cycle.ssm["design"],
+            cycle.ssm["transition"],
+            cycle.ssm["selection"],
+            cycle.ssm["state_cov"],
+        ],
+        mode="FAST_COMPILE",
+    )(params["cycle_length"], params["cycle_dampening_factor"], params["sigma_cycle"])
+
+    # each block is [1, 0] for design
+    expected_Z = np.zeros((3, 6))
+    expected_Z[0, 0] = 1.0
+    expected_Z[1, 2] = 1.0
+    expected_Z[2, 4] = 1.0
+    np.testing.assert_allclose(Z, expected_Z)
+
+    # each block is 2x2 frequency transition matrix for given cycle length (12 here),
+    # scaled by dampening factor
+    block = _frequency_transition_block(12, 1).eval() * params["cycle_dampening_factor"]
+    expected_T = np.zeros((6, 6))
+    for i in range(3):
+        expected_T[2 * i : 2 * i + 2, 2 * i : 2 * i + 2] = block
+    np.testing.assert_allclose(T, expected_T)
+
+    # each block is 2x2 identity for selection
+    expected_R = np.zeros((6, 6))
+    for i in range(3):
+        expected_R[2 * i : 2 * i + 2, 2 * i : 2 * i + 2] = np.eye(2)
+    np.testing.assert_allclose(R, expected_R)
+
+    # each block is sigma^2 * I_2 for state_cov
+    sigmas = params["sigma_cycle"]
+    expected_Q = np.zeros((6, 6))
+    for i in range(3):
+        expected_Q[2 * i : 2 * i + 2, 2 * i : 2 * i + 2] = np.eye(2) * sigmas[i] ** 2
+    np.testing.assert_allclose(Q, expected_Q)
+
+
+def test_add_multivariate_cycle_components_with_different_observed():
```

Comment:
@jessegrabowski , you wrote:
> Also please add a test adding a cycle component to another cycle component with a different observed state name. Check the resulting matrices come out as expected.

That's what I tried doing with this test, but LMK if that's overly complex, or even wrong, I wasn't super sure